tests/conftest.py

```diff
@@ -337,24 +337,34 @@ def generate_greedy_logprobs_limit(
                 processor_kwargs["images"] = images[i]
 
             inputs = self.processor(**processor_kwargs)
-            input_ids = inputs.input_ids
-
-            output = self.model.generate(
-                **self.wrap_device(inputs),
-                use_cache=True,
-                do_sample=False,
-                max_new_tokens=max_tokens,
-                output_hidden_states=True,
-                return_dict_in_generate=True,
-                **kwargs,
-            )
-
+            if "MiniCPM-Llama3-V-2_5" in self.model_name:
+                kwargs.pop("eos_token_id", None)
+                output = self.model.generate(
+                    self.wrap_device(inputs),
+                    use_cache=True,
+                    do_sample=False,
+                    max_new_tokens=max_tokens,
+                    output_hidden_states=True,
+                    return_dict_in_generate=True,
+                    tokenizer=self.tokenizer,
+                    **kwargs,
+                )
+            else:
+                output = self.model.generate(
+                    **self.wrap_device(inputs),
+                    use_cache=True,
+                    do_sample=False,
+                    max_new_tokens=max_tokens,
+                    output_hidden_states=True,
+                    return_dict_in_generate=True,
+                    **kwargs,
+                )
             seq_logprobs: List[torch.Tensor] = []
             for _, hidden_states in enumerate(output.hidden_states):
                 last_hidden_states = hidden_states[-1][0]
                 logits = torch.matmul(
-                    last_hidden_states,
-                    self.model.get_output_embeddings().weight.t(),
+                    last_hidden_states.clone(),
+                    self.model.get_output_embeddings().weight.t().clone(),
```

Comment:
Why do we need to clone this?

Reply:
Because, the generate of our code activate inference mode of torch. If not clone, there will raise an error like this:
![image](https://github.com/user-attachments/assets/135c10de-c1ef-40ba-b43f-a6c1365c3197)

Reply:
Can you put the `generate` code inside `torch.no_grad` to avoid this?

Reply:
You mean in `conftest.py` ？ Or in our code? Actually this happens when putting  in because https://huggingface.co/HwwwH/MiniCPM-Llama3-V-2_5/blob/main/modeling_minicpmv.py#L260. I can remove this line to avoid the error.

Reply:
Yeah, in `conftest.py`. It's better to fix this in the HF repo though, as you said.